dm/dm/worker/source_worker.go

```diff
@@ -798,35 +799,43 @@ func (w *SourceWorker) handleSubTaskStage(ctx context.Context, stageCh chan ha.S
 // operateSubTaskStage returns TaskOp.String() additionally to record metrics.
 func (w *SourceWorker) operateSubTaskStage(stage ha.Stage, subTaskCfg config.SubTaskConfig) (string, error) {
 	var op pb.TaskOp
-	switch {
-	case stage.Expect == pb.Stage_Running, stage.Expect == pb.Stage_Paused:
-		if st := w.subTaskHolder.findSubTask(stage.Task); st == nil {
-			// create the subtask for expected running and paused stage.
-			log.L().Info("start to create subtask", zap.String("sourceID", subTaskCfg.SourceID), zap.String("task", subTaskCfg.Name))
-
+	log.L().Info("operateSubTaskStage",
+		zap.String("sourceID", subTaskCfg.SourceID),
+		zap.String("task", subTaskCfg.Name),
+		zap.Stringer("stage", stage))
+
+	// for new added subtask
+	if st := w.subTaskHolder.findSubTask(stage.Task); st == nil {
+		switch stage.Expect {
+		case pb.Stage_Running, pb.Stage_Stopped:
+			log.L().Info("start to create subtask in operateSubTaskStage", zap.String("sourceID", subTaskCfg.SourceID), zap.String("task", subTaskCfg.Name))
 			expectValidatorStage, err := getExpectValidatorStage(subTaskCfg.ValidatorCfg, w.etcdClient, stage.Source, stage.Task, stage.Revision)
 			if err != nil {
 				return opErrTypeBeforeOp, terror.Annotate(err, "fail to get validator stage from etcd")
 			}
-
-			err = w.StartSubTask(&subTaskCfg, stage.Expect, expectValidatorStage, true)
-			return opErrTypeBeforeOp, err
-		}
-		if stage.Expect == pb.Stage_Running {
-			op = pb.TaskOp_Resume
-		} else if stage.Expect == pb.Stage_Paused {
-			op = pb.TaskOp_Pause
+			return opErrTypeBeforeOp, w.StartSubTask(&subTaskCfg, stage.Expect, expectValidatorStage, true)
+		default:
+			// not valid stage
+			return op.String(), w.OperateSubTask(stage.Task, op)
 		}
-	case stage.IsDeleted:
-		op = pb.TaskOp_Stop
+	}
+	// todo(ehco) remove  pause and resume after using openapi to impl dmctl
+	switch stage.Expect {
+	case pb.Stage_Stopped, pb.Stage_Paused:
```

Comment:
I found in many place stopped and paused are similar, can we only use paused?

Reply:
prefer unify to `stopped` as we name it in the new `dmctl`

Reply:
yes i will try merge `stopped` and `paused` later, i left a todo in https://github.com/Ehco1996/ticdc/blob/87a99e48fb422d9e1e76d5a749c1d09790dcac7d/dm/dm/worker/source_worker.go#L822